github/githubapp.go

```diff
@@ -17,18 +19,38 @@ type GitHubApp struct {
 	ClientId       string `json:"client_id"`
 	InstallationId int    `json:"installation_id"`
 	PrivateKey     string `json:"private_key"` // SENSITIVE
+	parsedRSAKey   *rsa.PrivateKey
 }
 
-func (app *GitHubApp) IsZero() bool {
-	return *app == GitHubApp{}
-}
+// Validate validates the GitHubApp configuration. Returns an error if
+// GitHubApp is misconfigured.
+func (app *GitHubApp) Validate() error {
+	var mandatory []string
 
-// generateJWTtoken returns a signed JWT token used to authenticate as GitHub App
-func generateJWTtoken(clientId, privateKey string) (string, error) {
-	key, err := jwt.ParseRSAPrivateKeyFromPEM([]byte(privateKey))
+	if app.ClientId == "" {
+		mandatory = append(mandatory, "github_app.client_id")
+	}
+	if app.InstallationId == 0 {
+		mandatory = append(mandatory, "github_app.installation_id")
+	}
+	if app.PrivateKey == "" {
+		mandatory = append(mandatory, "github_app.private_key")
+	}
+
+	if len(mandatory) > 0 {
+		return fmt.Errorf("github_app: missing mandatory keys: %s", strings.Join(mandatory, ", "))
```

Comment:
we first validate that all 3 fields are configured and then we parse the private RSA key...